components/dialog/dialog-mixin.js

```diff
@@ -231,12 +231,32 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 		if (this._opener && this._opener.focus) {
 			// wait for inactive focus trap
 			requestAnimationFrame(() => {
-				tryApplyFocus(this._opener);
+				let focusable = this._opener;
 				this._opener = null;
+				if (!isFocusable(focusable)) {
```

Comment:
This essentially re-creates `tryApplyFocus`.

Reply:
I think this would be more readable to keep it as a separate method, `this._tryApplyFocus()`.